### DISCUSSION TITLE
Inputs aus Onboarding AGI SZ

### DISCUSSION THREAD

**Discussion by olivergrimm** [geowerkstatt/geopilot#314] [General]
Die folgenden Punkte wurden im Rahmen des gestrigen Ob-Boardings des AGI SZ auf upload.geo.sz.ch festgestellt bzw. bemerkt:

- Bei INTERLIS-Mandaten sollte ein Prüfung des abgegeben Datensatzes in Bezug auf das erwartete Modell implementiert werden.
- Im STAC Browser wäre es dienlich, wenn der Katalogname mit der Organisation ergänzt würde, damit dieser auch bei gleich benannten Katalogen identifizierbar bleibt.
- Idee einer Benachrichtigung bei neuen Lieferungen.
- "Teillieferung" als Lieferungs-Eigenschaft entfernbar machen
- Konzeptionierung eines neuen Validierungsendpunktes gegenüber einem FME Workspace
- Eingabe von Lat/Lon in Admin UI umkehren, damit die Werte gemäss der Ausgabe auf map.geo.admin.ch eingegeben werden können.

**Comment by Philippluca** [comment]
Ich ergänze hier mal noch die Punkte, welche wir heute besprochen haben:

- Integrationen von 3. Systemen (FME, Abgabeprozesse etc. ) sollten einfacher geschehen.
- Möglicherweise gibt es 3. Systeme, welche nur auf die STAC-API der Applikation zugreiffen müssen. Diese brauchen nicht unbedingt ein personalisiertes Login & dürfen einen höheren Vertrauenslevel aufweisen.
- User sollten sich selbst registiren können. Keine "Einladung" in eien MS Entra tennant notwendig.

**Comment by KunoEpper** [comment]
Nachfolgend Überlegungen zum internen Datenmodell von geoPilot

### Hintergrund
Der Kanton SZ wird die Applikation für kommunale Datenlieferungen im ÖREB-Umfeld nutzen. Die Gemeinden liefern die Daten jedoch nicht selber, sondern beauftragen Büros, die Daten nachzuführen und abzugeben. Die Information, für welche Gemeinde eine Datenlieferung erfolgt, ist für uns wichtig.
Die Klasse **Organisation** wurde anfänglich als "Gemeinde" verstanden. Es sollten aber die Büros sein, welche von den Gemeinden für die Nachführung mandatiert wurden. Somit fehlt uns die wichtige Information, für welche Gemeinde eine Lieferung erfolgt.

### Lösungsversuch
Dies veranlasste mich, einen Modellierungsvorschlag auszuarbeiten, welcher unsere Situation abdeckt und evtl. auch für andere von Interesse sein könnte. In diesem Fall soll er als Diskussionsvorschlag für die Erweiterung von geoPilot dienen.

### Modell (allgemeine Fassung)
Eine Gemeinde beauftrag (bzw. mandatiert) eine Organisation, Daten nachzuführen.
Eine Organisation kann für mehrere Gemeinden Aufträge ausführen.
![Allgemeines - Datenmodell allgemein](https://github.com/user-attachments/assets/e9a97a5b-1162-450e-88d7-d49f7ac6bdde)

### Modell (detaillierte Fassung)
Die n:m-Beziehung zwischen **Gemeinde** und **Organisation** wird durch die Klasse **Mandat** aufgelöst.
Einem **Mandat** können **Lieferungen** zugeordnet werden, welche von **Usern** aufgegeben werden.
Dies entwpricht dem aktuellen Datenmodell von geoPilot entspricht (mit der Ausnahme einer Kardinalität).
![Allgemeines - Datenmodell vollständig](https://github.com/user-attachments/assets/9e8ed6e5-263c-4101-b284-e90c5c5baa87)

### nächste Schritte
Diese Modellierung würde unsere Situation perfekt ergänzen.
Wie sehen dies die anderen?

**Reply by Philippluca** [reply]
@KunoEpper Wenn ich es richtig verstanden habe, dann würde es mit der Umsetzung dieses Datenmodells weiterhin für jede Gemeinde und Thema ein Madat erstellt werden. Der Gewinn würde hier also nicht darin liegen, dass die Erfassung von Mandaten weniger aufwändig wird, sonder darin, dass jede Lieferung über das Mandat direkt einer Gemeinde zugeordnet werden kann. Das Mandat bereits heute einen Extent. Dieser würde das Mandat nach deiner Vorstellung nun einfach von der Gemeinde erben. So muss dieser nur 1x erfasst werden. 

Ich hätte hier einen Gegenvorschlag:

```mermaid
erDiagram
    Gemeinde {
        string Name
        string BFS-Nummer
        string Geometrie "Bounding-Box|exakt"
    }
    Mandat {
        string Name
        string[] Validierungsprofile
    }
    Organisation {
        string Name
    }
    Lieferung {
        string Kennung
    }
    User {
        string Name
    }
    Mandierung {
        date zBStart-Datum
        date zBEnd-Datum
        string Kontakt
    }




    Mandierung }o--|| Gemeinde: "vergeben durch"
    Mandierung }o--|| Mandat: "beschrieben durch"
    Mandierung }o--|| Organisation: "bearbeitet durch"
    Organisation }o--o{ User: "Angestellt durch"

    Lieferung }o--|| Mandierung: "getätigt im Ramen von"
    Lieferung }o--|| User: "getätigt durch"

```

**Comment by Philippluca** [comment]
Notizen zu Followup Inputs aus Onboarding AGI Schwyz (SZ):

*   Mandate pro Gemeinde sind zu aufwändig zu Erfassen
    *   (Gemeinde, Organisation, Modell/Kataster)
    *   Import aus CSV toll oder initial durch GeoW.
    *   Aktuell Fehlt eine Zwischenebene der Gemeinden, welche die Verantwortung Tragen
*   Organisation der Person/Mandat in STAC anzeigen.
*   STAC Browser muss Fragen beantworten können (Bessere Suche):
    *   Ist im Zeitram XY eine Lieferung für Gemeide / Operat passiert.
    *   Ist für Gemeinde X von Organisation Y eine Abgabe zum Mandat Z erfolgt.
    *   Suche nach Gemeinde passiert oft auch auf der Bezeichnung nicht dem Extent.
    *   Kuno Epper ist aktuell noch nicht 100% sicher, wie STAC angewendet wird, daher aktuell kein direktes Problem. Erst muss mal das Tool geopilot mit STAC angewendet und etabliert werden, um hier allenfalls sinnvolle Verbesserungen vorschlagen zu können.
*   Idee einer Benachrichtigung bei neuen Lieferungen.
    *   Aktuell keine direkte Relevanz. Kann mit z.B. FME im Integrationsprozess erledigt werden. Sollte es bei anderen Kunden/Anwendern Thema werden, ist auch SZ am Feature interessiert.
*   "Teillieferung" als Lieferungs-Eigenschaft soll optional werden.
    *   Konfigurative Option müsste da sein.
    *   Aktuell müssen die Kunden geschult werden, hier keine Angabe zu tätigen.
*   FME Validierung:
    *   FME Server Anbindung möglich.
    *   Input XTF & Metainformationen zur Lieferung (Organisation, User, Mandat, etc. )
    *   Prozess: INTERLIS-Validierung -> Deklaration Mandat -> FME-Validierung -> Finale Abgabe
    *   Zugriff auf FME Server und Definition der Workspace als "Validierung Profil"
    *   Pflege direkt auf FME Server. Ansteuerung direkt über Name oder optional Selektion nach Möglichkeit.
*   Eingabe von Lat/Lon in Admin UI umkehren, damit die Werte gemäss der Ausgabe auf [map.geo.admin.ch](http://map.geo.admin.ch/) eingegeben werden können.
    *   "Nice to Have". Aktuell bereits bei "SZ" in Excel gelöst. Würde Zukünftige Kunden wohl unterstützen.
    *   Karte aktuell nicht in Diskussion.
*   FME Desktop Anbindung noch nicht getestet.
    *   SZ meldet sich, falls es noch weitere Probleme gibt.
*   Dass man Benutzer selbst erfassen kann & Deauthorisieren kann wäre erwartet.
    *   Es ist eine Erwartung 
    *   Erlaubt es die Pflege direkt anzugehen, ohne den Medienbruch "E-Mail".
